torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -66,25 +66,65 @@ TensorView* newOutputTV(const std::vector<Val*>& vals, DataType dtype) {
       out_domain[i] = new IterDomain(dom[i]->start(), dom[i]->extent());
     }
   }
-
-  std::transform(
-      out_domain.begin(),
-      out_domain.end(),
-      out_domain.begin(),
-      [](IterDomain* dom) {
-        if (dom == nullptr)
-          return new IterDomain(
-              new Int(0), new Int(1), ParallelType::Serial, false, false, true);
-        return dom;
-      });
+  for (size_t dim_i = 0; dim_i < out_domain.size(); dim_i++) {
+    if (out_domain[dim_i] == nullptr) {
+      BroadcastType bcast_type = BroadcastType::WithoutStride;
+      for (size_t inp_i = 0; inp_i < tvs.size(); inp_i++) {
+        auto dim =
+            TensorDomain::noReductions(tvs[inp_i]->getRootDomain())[dim_i];
+        // If there's an unresolved bcast dim and it came from a strided dim,
+        // assume output of it should be strided too
+        if (dim->getBroadcastType() == BroadcastType::WithStride) {
+          bcast_type = BroadcastType::WithStride;
+          break;
+        }
+      }
+      out_domain[dim_i] = new IterDomain(
+          new Int(0),
+          new Int(1),
+          ParallelType::Serial,
+          false,
+          false,
+          bcast_type);
+    }
+  }
 
   return new TensorView(new TensorDomain(out_domain), dtype);
 }
 
-Val* newOutputVal(const std::vector<Val*>& vals) {
-  TORCH_INTERNAL_ASSERT(
-      !vals.empty(), "Cannot promote values if there aren't any.");
+std::vector<Val*> maybeBroadcast(const std::vector<Val*>& vals) {
```

Comment:
is this a local helper?

Reply:
thanks. it's really hard to tell while reviewing a PR (which is a reason I prefer static instead of anonymous namespaces)